jina/clients/mixin.py

```diff
@@ -29,6 +31,68 @@ def _include_results_field_in_param(parameters: Optional['Dict']) -> 'Dict':
     return parameters
 
 
+class MutateMixin:
+    """The GraphQL Mutation Mixin for Client and Flow"""
+
+    def mutate(
+        self,
+        mutation: str,
+        variables: Optional[dict] = None,
+        timeout: Optional[float] = None,
+        headers: Optional[dict] = None,
+    ):
+        """Perform a GraphQL mutation
+
+        :param mutation: the GraphQL mutation as a single string.
+        :param variables: variables to be substituted in the mutation. Not needed if no variables are present in the mutation string.
+        :param timeout: HTTP request timeout
+        :param headers: HTTP headers
+        :return: dict containing the optional keys ``data`` and ``errors``, for response data and errors.
+        """
+        with ImportExtensions(required=True):
+            from sgqlc.endpoint.http import HTTPEndpoint as SgqlcHTTPEndpoint
+
+            proto = 'https' if self.args.https else 'http'
+            graphql_url = f'{proto}://{self.args.host}:{self.args.port}/graphql'
+            endpoint = SgqlcHTTPEndpoint(graphql_url)
+            res = endpoint(
+                mutation, variables=variables, timeout=timeout, extra_headers=headers
+            )
+            return res
+
+
+class AsyncMutateMixin(MutateMixin):
+    """The async GraphQL Mutation Mixin for Client and Flow"""
+
+    async def _async_mutate(
+        self,
+        mutation: str,
+        variables: Optional[dict] = None,
+        timeout: Optional[float] = None,
+        headers: Optional[dict] = None,
+    ):
+        return super().mutate(mutation, variables, timeout, headers)
+
+    async def mutate(
+        self,
+        mutation: str,
+        variables: Optional[dict] = None,
+        timeout: Optional[float] = None,
+        headers: Optional[dict] = None,
+    ):
+        """Perform a GraphQL mutation, asynchronously
+
+        :param mutation: the GraphQL mutation as a single string.
+        :param variables: variables to be substituted in the mutation. Not needed if no variables are present in the mutation string.
+        :param timeout: HTTP request timeout
+        :param headers: HTTP headers
+        :return: dict containing the optional keys ``data`` and ``errors``, for response data and errors.
+        """
+        return await asyncio.get_event_loop().run_in_executor(
```

Comment:
Let's use `jina.helper.get_or_reuse_loop()` here instead of `asyncio.get_event_loop()`.